src/signing_commitment.rs

```diff
@@ -6,32 +6,79 @@ use std::io;
 
 use reddsa::frost::redjubjub::round1::NonceCommitment;
 
-use crate::participant::{Identity, IDENTITY_LEN};
+use crate::participant::{Identity, Secret, Signature, SignatureError, IDENTITY_LEN};
 
 const NONCE_COMMITMENT_LENGTH: usize = 32;
-pub const SIGNING_COMMITMENT_LENGTH: usize = IDENTITY_LEN + NONCE_COMMITMENT_LENGTH * 2;
+pub const SIGNATURE_DATA_LENGTH: usize = IDENTITY_LEN + NONCE_COMMITMENT_LENGTH * 2;
+pub const SIGNING_COMMITMENT_LENGTH: usize = SIGNATURE_DATA_LENGTH + Signature::BYTE_SIZE;
+
+fn signature_data(
+    identity: &Identity,
+    hiding: &NonceCommitment,
+    binding: &NonceCommitment,
+) -> [u8; SIGNATURE_DATA_LENGTH] {
+    let mut data = [0u8; SIGNATURE_DATA_LENGTH];
+    let parts = [
+        &identity.serialize()[..],
+        &hiding.serialize(),
+        &binding.serialize(),
+    ];
+    let mut slice = &mut data[..];
+    for part in parts {
+        slice[..part.len()].copy_from_slice(part);
+        slice = &mut slice[part.len()..];
+    }
+    assert_eq!(slice.len(), 0);
```

Comment:
Nit: `debug_assert_eq`

Reply:
Updated, but do we need to update [this](https://github.com/iron-fish/ironfish-frost/blob/main/src/participant.rs#L56) too?

Reply:
Yeah, that'd be nice